ui/src/shared/utils/analytics.ts

```diff
@@ -23,3 +25,25 @@ export const fireUserDataReady = function fireUserDataReady(
     },
   })
 }
+
+export const fireGAEvent = (event: string, payload: object = {}) => {
+  window.dataLayer.push({
+    event,
+    ...payload,
+  })
+}
+
+export const fireQueryEvent = (ownOrg: string, queryOrg: string) => {
+  if (ownOrg === queryOrg) {
+    fireGAEvent('orgData_queried')
+  } else {
+    fireGAEvent('demoData_queried')
+  }
+}
+
+export const fireDashboardViewedEvent = (dashboardName: string) => {
+  const demo_dataset = DemoDataDashboardNames[dashboardName]
+  if (demo_dataset) {
+    fireGAEvent('demoData_dashboardViewed', {demo_dataset})
+  }
```

Comment:
is this event only valuable with the demo dataset? or could something like this also be of value:

```ts
else {
  fireGAEvent('demoData_dashboardViewed')
}
```

Reply:
also, i just noticed this is JS code, camelCase that variable name!

Reply:
I'm trying to keep this PR to only the things that were requested for demo data, when we actually instrument GA, we can take a wholistic view of all the events we should track and how.